Source/monster.cpp

```diff
@@ -1949,50 +1949,51 @@ void GroupUnity(int i)
 	assert(i >= 0 && i < MAXMONSTERS);
 	auto &monster = Monsters[i];
 
-	if (monster.leaderflag == MonsterRelation::Individual) {
+	if (monster.Relation == MonsterRelation::Individual)
 		return;
-	}
 
-	if (monster.leader != 0) {
-		auto &leader = Monsters[monster.leader];
-		if (IsLineNotSolid(monster.position.tile, leader.position.future)) {
-			if (monster.leaderflag == MonsterRelation::Leader && monster.position.tile.WalkingDistance(leader.position.future) < 4) {
-				leader.packsize++;
-				monster.leaderflag = MonsterRelation::Minion;
-			}
-		} else {
-			leader.packsize--;
-			monster.leaderflag = MonsterRelation::Leader;
-		}
+	assert(monster.leader >= 0);
 
-		if (monster.leaderflag == MonsterRelation::Minion) {
-			if (monster._msquelch > leader._msquelch) {
-				leader.position.last = monster.position.tile;
-				leader._msquelch = monster._msquelch - 1;
-			}
+	auto &leader = Monsters[monster.leader];
+	if (IsLineNotSolid(monster.position.tile, leader.position.future)) {
+		if (monster.Relation == MonsterRelation::SeparatedMinion
+		    && monster.position.tile.WalkingDistance(leader.position.future) < 4) {
+			// Reunite the separated monster with the pack
+			leader.packsize++;
+			monster.Relation = MonsterRelation::Minion;
 		}
+	} else if (monster.Relation == MonsterRelation::SeparatedMinion) {
+		// Monster is already separated from pack
+	} else {
+		leader.packsize--;
+		monster.Relation = MonsterRelation::SeparatedMinion;
 	}
 
-	if (monster._mAi == AI_GARG && (monster._mFlags & MFLAG_ALLOW_SPECIAL) != 0) {
-		if (monster.leaderflag == MonsterRelation::Minion || monster.packsize > 0) {
-			monster._mFlags &= ~MFLAG_ALLOW_SPECIAL;
-			monster._mmode = MM_SATTACK;
+	if (monster.Relation == MonsterRelation::Minion) {
+		if (monster._msquelch > leader._msquelch) {
+			leader.position.last = monster.position.tile;
+			leader._msquelch = monster._msquelch - 1;
 		}
+		if (leader._mAi == AI_GARG && (leader._mFlags & MFLAG_ALLOW_SPECIAL) != 0) {
+			leader._mFlags &= ~MFLAG_ALLOW_SPECIAL;
+			leader._mmode = MM_SATTACK;
+		}
+		return;
 	}
 
-	if (monster.leaderflag == MonsterRelation::Leader && monster._uniqtype != 0) {
-		if ((UniqMonst[monster._uniqtype - 1].mUnqAttr & 2) == 0) {
-			return;
+	if (monster._uniqtype == 0 || (UniqMonst[monster._uniqtype - 1].mUnqAttr & 2) == 0)
+		return;

```

Comment:
Uniques are never also minions so the code past this point would be unused afaik?

Reply:
Hmm apparently I'm wrong about this as well in some cases...

Reply:
Did a small hack to spawn all unique monsters and indeed nothing ever gets past this point. But I did bump in to some interesting unused unique monsters :)

Here is a few of them:
![image](https://user-images.githubusercontent.com/204594/126077756-ce9b590f-60c5-4070-9a74-bf22c3e1dd33.png)
![image](https://user-images.githubusercontent.com/204594/126077768-39f12e00-b319-462a-a12d-6c1309811a3e.png)
![image](https://user-images.githubusercontent.com/204594/126077786-814a3caa-0e74-4fbf-aa5f-5a157caa2d84.png)
![image](https://user-images.githubusercontent.com/204594/126077799-acd11ba4-eb68-422d-a8cc-99adae620e9f.png)

Reply:
I also looked again at the code and think that's impossible.
So I removed the code and added an assert. 👍 

btw: sometimes I think we should add a flag in the ini to enable all the missing content. similar to bard etc. But hey we still have some versions of devilutionX that will come in the future. 😁

Reply:
Good thinking with the assert.